win32/scalyr-agent.spec

```diff
@@ -0,0 +1,112 @@
+# -*- mode: python ; coding: utf-8 -*-
+# Copyright 2014-2020 Scalyr Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from PyInstaller.utils.hooks import collect_submodules
+
+
+block_cipher = None
+
+
+main_a = Analysis(['source_root\\scalyr_agent\\agent_main.py'],
+             pathex=[
+                'source_root\\scalyr_agent\\builtin_monitors',
+				'source_root\\scalyr_agent\\third_party',
+				'source_root\\scalyr_agent\\third_party_python2'
+				],
+			 hiddenimports = collect_submodules('scalyr_agent.builtin_monitors') + ['pkg_resources.py2_warn'],
+             datas=[('data_files\\VERSION', '.'), ('data_files\\licenses', 'third_party_licenses')],
+)
+
+config_a = Analysis(['source_root\\scalyr_agent\\config_main.py'],
+             pathex=[
+                'source_root\\scalyr_agent\\builtin_monitors',
+                'source_root\\scalyr_agent\\third_party',
+				'source_root\\scalyr_agent\\third_party_python2'
+				],
+             hiddenimports = collect_submodules('scalyr_agent.builtin_monitors') + ['pkg_resources.py2_warn'],
+             datas=[('data_files\\VERSION', '.'), ('data_files\\licenses', 'third_party_licenses')],
+             )
+
+service_a = Analysis(['source_root\\scalyr_agent\\platform_windows.py'],
+             pathex=[
+                'source_root\\scalyr_agent\\builtin_monitors',
+                'source_root\\scalyr_agent\\third_party',
+				'source_root\\scalyr_agent\\third_party_python2',
+				],
+             hiddenimports = collect_submodules('scalyr_agent.builtin_monitors') + ['pkg_resources.py2_warn'],
+             datas=[('data_files\\VERSION', '.'), ('data_files\\licenses', 'third_party_licenses')],
+             )
+
+
+
+MERGE( (main_a, 'scalyr-agent-2', 'scalyr-agent-2.exe'), (config_a, 'scalyr-agent-2-config', 'scalyr-agent-2-config.exe'), (service_a, 'ScalyrAgentService', 'ScalyrAgentService.exe'))
```

Comment:
Can you please also make sure we include ``build_info`` file and install it in the same directory where the agent is installed?

This file is created as part of ``build_package.py`` build pricess and it contains information about the git revision which was used to build the package.

(Might also need some changes to also create that file for Windows builds, but hopefully it shouldn't be too hard)